packages/services/f-feature-management/test/featureGetter.test.js

```diff
@@ -0,0 +1,118 @@
+import { getFeature, init, poll } from '../src/featureGetter';
+import fetchMock from "jest-fetch-mock";
+
+fetchMock.enableMocks();
+
+const settings = {
+  scope: 'test-scope',
+  environment: 'test-env',
+  key: 'test-key'
+};
+
+const fetchResponse = {
+  createdAt: '2021-09-10 15:00',
+  features: [
+    {
+      key: 'key1',
+      testVal: 'val1'
+    }
+  ]
+};
+
+const flushPromises = () => new Promise(res => process.nextTick(res))
+
+describe('When calling poll', () => {  
+
+  beforeEach(async () => {
+    fetch.resetMocks();
+    jest.useFakeTimers();
+
+    fetch.mockResponse(JSON.stringify(fetchResponse));
+
+    await poll(settings);    
+  });
+
+  it('Makes an initial call to fetch', async () => {
+    
+    expect(fetch).toHaveBeenCalledTimes(1);
+    expect(fetch).toHaveBeenCalledWith(`https://features.api.justeattakeaway.com/config/v1/${settings.scope}/${settings.environment}-${settings.key}`);
+  });
+
+  it('Starts polling', async () => {
+    expect(setInterval).toHaveBeenCalledTimes(1);
+    expect(setInterval).toHaveBeenCalledWith(expect.any(Function), 30000);
+
+    jest.advanceTimersByTime(29000);
+    expect(fetch).toHaveBeenCalledTimes(1);    
+
+    jest.advanceTimersByTime(1000);
```

Comment:
Feels like there could be a potential for this to be flake, might be worth adding some more ms?

Reply:
The test below looks a bit better, with `35 + 30 > 60` seconds, whereas here we have `29 + 1 = 30`

Reply:
It won’t be flaky. Computers are very precise about these things! But I could add some time to prevent any doubt on the part of developers, who are less precise.

Reply:
Done